add methods for collapse rnaseq and immunedeconv (#64)
AlexsLemonade/OpenPBTA-manuscript: content/03.methods.md

We also added a second method of quantification using kallisto [@doi:10.1038/nbt.3519] v0.43.1.
This method differs in that it uses pseudoaligments using fastq reads directly to the aforementioned GENCODE 27 reference.

### Gene Expression Matrices with Unique HUGO Symbols

> Algorithms that perform gene set enrichment, molecular subtyping or immune-profiling require a RNA-seq gene expression matrix as input, with HUGO gene symbols as row names and sample names as column names. This is however not straight-forward to achieve as there are a small proportion of gene symbols which are mapped to multiple Ensembl gene identifiers. In Gencode v27, there are 212 gene symbols mapped to 1866 Ensembl gene identifiers. 

Reviewer: ```suggestion
Algorithms that perform gene set enrichment, molecular subtyping, or immune-profiling, for example, require an RNA-seq gene expression matrix as input, with HUGO gene symbols as row names and sample names as column names. 
This is not straight-forward, as there are a small proportion of gene symbols that map to multiple Ensembl gene identifiers (in Gencode v27, 212 gene symbols map to 1866 Ensembl gene identifiers). 
```

@komalsrathi thanks for this! Starting to review, but can you separate each sentence on its own line and remove the line spacing between paragraphs? Made a few minor edits above to condense.

> To create a matrix with de-duplicated gene symbols, we first filtered zero-expression genes by using the cut-off of FPKM > 0 in at least 1 sample across the PBTA cohort. Taking the expressed multi-mapped genes, we found 168 genes mapped to 1393 Ensembl gene identifiers in the stranded dataset and 139 genes mapped to 429 Ensembl gene identifiers in the polyA dataset. Of these, only a small proportion were found to be protein coding genes i.e. 3% (42/1393) in stranded and 10% (42/429) in the poly-A dataset. In fact, the top two most represented gene biotypes were misc_RNA (60% for stranded; 36% for poly-A) and snoRNA (24% for stranded and 27% for poly-A).

Reviewer: ```suggestion
We first removed genes with no expression using a cut-off of FPKM > 0 in at least 1 sample across the PBTA cohort. Next, of the expressed multi-mapped genes, we found 168 genes mapped to 1393 Ensembl gene identifiers in the stranded dataset and 139 genes mapped to 429 Ensembl gene identifiers in the polyA dataset. 
Of these, only a small proportion were found to be protein coding genes, i.e. 3% (42/1393) in stranded and 10% (42/429) in the poly-A dataset. 
The top two most represented gene biotypes were misc_RNA (60% for stranded; 36% for poly-A) and snoRNA (24% for stranded and 27% for poly-A).
```

I wonder if we should include the exact details of this, as it may change once we get the additional stranded RNA-Seq data, or just make a note that this may need to be updated. CC: @jaclyn-taroni

> Following this, we computed the mean FPKM across all samples per gene. For each multi-mapped gene symbol, we chose the Ensembl identifier corresponding to the maximum mean FPKM. The idea here was to choose the identifier that best represented the expression of the gene. After collapsing, there were a total of 46400 unique expressed genes in the poly-A dataset and a total of 53011 unique expressed genes remaining in the stranded dataset

Reviewer: ```suggestion
We computed the mean FPKM across all samples per gene and for each multi-mapped gene symbol, we chose the Ensembl identifier corresponding to the maximum mean FPKM with the goal of choosing the identifier that best represented the expression of the gene. 
After collapsing gene identifiers, there were a total of 46,400 unique expressed genes in the poly-A dataset and a total of 53,011 unique expressed genes remaining in the stranded dataset.
```
